src/store/reset-password/saga.ts

```diff
@@ -0,0 +1,52 @@
+import { resetPassword as resetPasswordApi } from './api';
+import { setLoading, setErrors, SagaActionTypes, setEmailSubmitted } from './index';
+import { ResetPasswordErrors } from '.';
+import { call, put, take } from 'redux-saga/effects';
+
+export function validateResetPassword({ email }) {
+  const validationErrors = [];
+
+  if (!email.trim()) {
+    validationErrors.push(ResetPasswordErrors.EMAIL_REQUIRED);
+  }
+
+  return validationErrors;
+}
+
+export function* resetPasswordPage() {
+  let success;
+  do {
+    const action = yield take(SagaActionTypes.ResetPassword);
+    yield put(setLoading(true));
+    yield put(setEmailSubmitted(false));
+
+    const validationErrors = validateResetPassword(action.payload);
+
+    if (validationErrors.length > 0) {
+      yield put(setErrors(validationErrors));
+      success = false;
+      continue;
+    }
+
+    const result = yield call(resetPasswordApi, action.payload);
```

Comment:
I think we need a try/catch somewhere in here because it's possible for the api request to outright fail and that would throw an exception.